lib/logstash/outputs/scalyr.rb

```diff
@@ -363,7 +375,15 @@ def multi_receive(events)
           }
           exc_sleep += sleep_interval
           exc_retries += 1
-          retry if @running
+          retry if @running and exc_retries < @max_retries
+          message = "Failed to send event after #{exc_retries} tries."
+          @logger.error(message, :error_data => exc_data, :retries => exc_retries, :sleep_time => exc_sleep)
+          if dlq_enabled? and @dlq_writer
```

Comment:
If it's disabled we should probably at least log a message - something along the lines of dropping X events after Y retries and perhaps small sample of dropped events.